libr/anal/types.c

```diff
@@ -277,3 +245,33 @@ R_API char *r_anal_type_format (RAnal *anal, const char *t) {
 	}
 	return NULL;
 }
+// Function prototypes api
+R_API int r_anal_type_func_exist(RAnal *anal, const char *func_name) {
+	char *fcn = sdb_const_get (anal->sdb_types, func_name, 0);
+	return fcn && !strcmp (fcn, "func");
+}
+
+R_API const char *r_anal_type_func_cc(RAnal *anal, const char *func_name) {
+	char *query = sdb_fmt (-1, "func.%s.cc", func_name);
+	return sdb_const_get (anal->sdb_types, query, 0);
+}
+
+R_API int r_anal_type_func_args_count(RAnal *anal, const char *func_name) {
+	char *query = sdb_fmt (-1, "func.%s.args", func_name);
+	return sdb_num_get (anal->sdb_types, query, 0);
+}
+
+R_API char *r_anal_type_func_args_type(RAnal *anal, const char *func_name, int i) {
+	char *query = sdb_fmt (-1, "func.%s.arg.%d", func_name, i);
+	char *ret = sdb_get (anal->sdb_types, query, 0);
+	char *comma = strchr (ret, ',');
+	*comma = 0;
+	return ret;
+
+}
+
+R_API char *r_anal_type_func_args_name(RAnal *anal, const char *func_name, int i) {
+	char *query = sdb_fmt (-1, "func.%s.arg.%d", func_name, i);
+	const char *ret = sdb_const_get (anal->sdb_types, query, 0);
+	return strchr (ret, ',') + 1;
```

Comment:
same as before this is automatic output of tcc it will never be null if so it is then invalid syntax